feat(install.sh): handle installation directory prompt responses case-insensitively

Daniel-Escamilla opened the pull request:
#### Description

Improve input handling in `install/install.sh` for confirmation prompts. Specifically:

- Interprets user responses case-insensitively to correctly handle both affirmative and negative answers.
- Adds clearer error messages when the user aborts input with `Ctrl+C` or `Ctrl+D`.
- Ensures that if the user declines installation in the default folder, the prompt exits gracefully with appropriate messaging.

#### Motivation and Context

This change ensures a more robust and user-friendly install experience, particularly when users provide lowercase or uppercase variants of "no" (`n`, `N`, `no`, `NO`) or interrupt input unexpectedly. Previously, such responses could lead to unclear errors or unintended behaviour.

Closes #5608 

#### How Has This Been Tested?

Manually tested the following cases in Starship:

- User inputs `y`, `Y`, `yes`, `YES` → script proceeds as expected.
- User inputs `n`, `N`, `no`, `NO` → script prompts again for an alternative installation path.
- User presses `Ctrl+C` → script exits cleanly.
- User presses `Ctrl+D` → script exits with error message.

#### Checklist:

- [x] I have tested using **Linux**
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

cappyzawa commented on install/install.sh:
```diff
@@ -257,21 +257,43 @@ detect_target() {
+        rc=$?
+        set -e
+        BIN_DIR=$path
+        if [ $rc != 0 ] || [ -z "$path"]; then
+          if [ -z "$path"]; then
```
Missing space before `]` - should be `[ -z "$path" ];`

cappyzawa commented on install/install.sh:
```diff
@@ -257,21 +257,43 @@ detect_target() {
-
 confirm() {
   if [ -z "${FORCE-}" ]; then
+    trap 'echo; error "Interrupted (please re-run with the '--yes' option)"; exit 130' INT
```
nits: This trap setup is duplicated on L276. Consider using a helper function to avoid repetition.

cappyzawa reviewed (commented):
Great fix for Issue #5608! The functionality works perfectly once the syntax errors are fixed.

Daniel-Escamilla commented:
Thanks for the review, @cappyzawa.

I've fixed the spacing issue in the condition and extracted the duplicated "trap" line into a helper function as suggested.
Let me know if there's anything else you'd like me to adjust.

cappyzawa commented:
Perfect! Thank you for addressing all the feedback. The syntax errors are fixed and the code is much cleaner with the `set_interrupt_handler()` function.

Functionality works exactly as expected - great fix for Issue #5608!

**One more UX improvement idea:** Just realized the custom path prompt could be even more user-friendly:

```bash
# Current
? Install Starship latest to: 

# What if we added some guidance?
Please enter the directory path where you want to install Starship (e.g., /usr/local/bin):
? Install Starship latest to: 
```

This would provide clearer guidance about what to enter. What do you think?

Everything else looks great! 🚀

Daniel-Escamilla commented:
Thanks for the suggestion — totally agreed it would help UX.

Just pushed the update with the improved prompt.
Let me know if anything else comes to mind

cappyzawa commented on install/install.sh:
```diff
@@ -257,21 +257,48 @@ detect_target() {
+      if [ "$yn" = "n" ] || [ "$yn" = "no" ]; then
+        set_interrupt_handler
+        set +e
+        printf "${MAGENTA}?${NO_COLOR} Where would you like to install Starship? ${GREY}(e.g., /usr/local/bin)${NO_COLOR}\n"
+        read -p "${MAGENTA}?${NO_COLOR} Install Starship ${GREEN}${VERSION}${NO_COLOR} to: " path
```
The first line is more of an explanation than a question, and for input prompts `>` is more natural than `?`:

```suggestion
        printf "Where would you like to install Starship? ${GREY}(e.g., /usr/local/bin)${NO_COLOR}\n"
        read -p "> Install Starship ${GREEN}${VERSION}${NO_COLOR} to: " path
```

This creates a cleaner flow:
```
  Where would you like to install Starship? (e.g., /usr/local/bin)
  > Install Starship latest to: |
```

davidkna commented:
Thanks for the contribution @Daniel-Escamilla, and thanks for reviewing @cappyzawa!
